common/lib/dependabot/version.rb

```diff
@@ -1,21 +1,36 @@
-# typed: true
+# typed: strong
 # frozen_string_literal: true
 
+require "sorbet-runtime"
+
 module Dependabot
   class Version < Gem::Version
+    extend T::Sig
+    extend T::Helpers
+
+    abstract!
+
+    sig { overridable.params(version: T.any(String, Dependabot::Version)).void }
     def initialize(version)
-      @original_version = version
+      @original_version = T.let(version.to_s, String)
```

Comment:
The only use of this is `to_semver`, and the only use of that is `ignore_condition`, which expects a `String`:

https://github.com/dependabot/dependabot-core/blob/389cb8a49af6a4c499da6322a6c138f734ccc310/common/lib/dependabot/config/ignore_condition.rb#L60-L73